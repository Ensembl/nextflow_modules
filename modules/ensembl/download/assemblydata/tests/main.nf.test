// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test download/assemblydata
nextflow_process {

    name "Test Process DOWNLOAD_ASSEMBLYDATA"
    script "../main.nf"
    process "DOWNLOAD_ASSEMBLYDATA"

    tag "modules"
    tag "modules_nfcore"
    tag "download"
    tag "download/assemblydata"

    test("Download assemblydata [stub]") {
        when {

            options "-stub"

            process {
                """
                input[0] = [[ id:'test']]
                """
            }
        }


        then {
            assert snapshot(
                    process.out.min_set.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("Minimal_file_set")

            assert snapshot(
                    process.out.opt_set.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("Optional_file_set")

            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.min_set.size() == 1 },
                { assert process.out.opt_set.size() == 1 },
                { assert process.out.min_set != null },
                { assert process.out.opt_set != null },
            )
        }
    }
}
