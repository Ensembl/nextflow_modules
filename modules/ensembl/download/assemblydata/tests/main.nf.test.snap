{
    "Optional_file_set": {
        "content": [
            [
                [
                    {
                        "accession": "test"
                    },
                    "GCA_017607445.1_ASM1760744v1_genomic.gff.gz",
                    "GCA_017607445.1_ASM1760744v1_protein.faa.gz",
                    "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T16:18:39.369126462"
    },
    "Download assemblydata [stub]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "test"
                        },
                        "GCA_017607445.1_ASM1760744v1_assembly_report.txt:md5,4b1bc7091004b82da5c28e484a3ebecc",
                        "GCA_017607445.1_ASM1760744v1_genomic.fna.gz:md5,af772cfb14e26609b7c215690baf2c3d",
                        "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz:md5,f4dac2b4d4e10cb3d247640350112f82"
                    ]
                ],
                "1": [
                    [
                        {
                            "accession": "test"
                        },
                        "GCA_017607445.1_ASM1760744v1_genomic.gff.gz:md5,6c58a33f10fcb1d502e4ade1eae87c4d",
                        "GCA_017607445.1_ASM1760744v1_protein.faa.gz:md5,af08975898232e57b75cc5f2617dbb16",
                        "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz:md5,f4dac2b4d4e10cb3d247640350112f82"
                    ]
                ],
                "2": [
                    "versions.yml:md5,397f2cc581ecbc8039f6ba1e998cadb3"
                ],
                "min_set": [
                    [
                        {
                            "accession": "test"
                        },
                        "GCA_017607445.1_ASM1760744v1_assembly_report.txt:md5,4b1bc7091004b82da5c28e484a3ebecc",
                        "GCA_017607445.1_ASM1760744v1_genomic.fna.gz:md5,af772cfb14e26609b7c215690baf2c3d",
                        "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz:md5,f4dac2b4d4e10cb3d247640350112f82"
                    ]
                ],
                "opt_set": [
                    [
                        {
                            "accession": "test"
                        },
                        "GCA_017607445.1_ASM1760744v1_genomic.gff.gz:md5,6c58a33f10fcb1d502e4ade1eae87c4d",
                        "GCA_017607445.1_ASM1760744v1_protein.faa.gz:md5,af08975898232e57b75cc5f2617dbb16",
                        "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz:md5,f4dac2b4d4e10cb3d247640350112f82"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,397f2cc581ecbc8039f6ba1e998cadb3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T16:18:39.547798388"
    },
    "Minimal_file_set": {
        "content": [
            [
                [
                    {
                        "accession": "test"
                    },
                    "GCA_017607445.1_ASM1760744v1_assembly_report.txt",
                    "GCA_017607445.1_ASM1760744v1_genomic.fna.gz",
                    "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T16:18:39.291805979"
    }
}